types/hippie/index.d.ts

```diff
@@ -0,0 +1,62 @@
+// Type definitions for hippie 0.5
+// Project: https://github.com/vesln/hippie
+// Definitions by: Nico Muschert <https://github.com/nicomuschert>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+// TypeScript Version: 2.2
+/// <reference types="node" />
+
+import http = require('http');
+
+interface Err {
+  AssertionError: string;
+  message: string;
+  showDiff: boolean;
+  actual: boolean;
+  expected: boolean;
+}
+
+interface EndCb {
+  (err: undefined|Err|Error, res: http.IncomingMessage, body?: object|string): void;
+}
+
+interface Opts {
+  [key: string]: string|{};
+}
+
+declare function hippie(app?: any): typeof hippie;
```

Comment:
I don't think this is what you want to export. Specifically, this allows users to directly access functions off the module:
```
var hippie = require('hippie');
hippie.end(...)
```
One easy way to fix this is to rename this `hippie` function to something else and export it (and still keep the namespace named `hippie`). Another option is to do something similar but use interfaces to avoid the `typeof`s:

```
interface hippie {
    del(url: string): hippie;
}

declare function hippieConstructor(): hippie;
export = hippieConstructor;
```